src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -107,9 +107,26 @@ public bool CanInline(MethodDesc caller, MethodDesc callee)
                 }
             }
 
+            EcmaModule rootModule = (root.OwningType as MetadataType)?.Module as EcmaModule;
+            EcmaModule calleeModule = (callee.OwningType as MetadataType)?.Module as EcmaModule;
+
+            // If this inline crosses module boundaries, ensure the modules agree on exception wrapping behavior.
```

Comment:
I think this should be directly in CorInfoImpl.cs. It is not a policy decision whether the inlining is allowed for this case. It is required for correctness, so it should not be abstracted by the policy.

Reply:
Ok. I can then undo some of the other changes, since the policy will no longer need to know the root method.